src/addrman.h

```diff
@@ -593,6 +577,8 @@ class CAddrMan
         Check();
     }
 
+    const std::vector<bool>& GetAsmap() const { return m_asmap; }
```

Comment:
please add LIFETIMEBOUND when returning a pointer to memory in `this`

Reply:
Good idea.

I actually plan to fully encapsulate asmap in its own component in #22910, so we're not passing out references to private data.